tests/SimplifyTest.elm

```diff
@@ -6349,6 +6349,35 @@ a = List.append [b] [c,d,0]
 a = [b,c,d,0]
 """
                         ]
+        , test "should report List.append <| on two list literals" <|
+            \() ->
+                """module A exposing (..)
+a = List.append [b] <| [c,d,0]
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "Appending literal lists could be simplified to be a single List"
+                            , details = [ "Try moving all the elements into a single list." ]
+                            , under = "List.append"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a = [b,c,d,0]
+"""
+                        ]
+        , test "should report List.append |> on two list literals" <|
+            \() ->
+                """module A exposing (..)
+a = [c,d,0] |> List.append [b]
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "Appending literal lists could be simplified to be a single List"
+                            , details = [ "Try moving all the elements into a single list." ]
+                            , under = "List.append"
+                            }
+                        ]
```

Comment:
Why do we not have a fix for this one? :thinking:

Reply:
I describe this in the PR body:
The elements that should be later in the combined list are located before when using `|>`, so
> There is no way to fix this without being able to extract source code of the piped list, therefore no fixes will be provided.

Reply:
Ah sorry, my bad, I didn't read it, went straight for the commits. But we do have a mechanism for that :)
https://package.elm-lang.org/packages/jfmengels/elm-review/latest/Review-Rule#withSourceCodeExtractor

We already use that, so that is the `extractSourceCode` field in the `Context`., You can grab the code in between the brackets and put them at the end.

Reply:
Yes, let's add the `sourceCodeExtractor` to the CheckInfo, that sounds good :+1: